packages/account/src/Components/trading-assessment/trading-assessment-form.jsx

```diff
@@ -0,0 +1,201 @@
+        current_question_index: 0,
+        current_question: {},
+    });
+    const [form_data, setFormData] = React.useState({});
+
+    const last_question_index = assessment_questions.length - 1;
+
+    React.useEffect(() => {
+        setCurrentQuestionDetails(prevState => ({
+            ...prevState,
+            current_question: assessment_questions[prevState.current_question_index],
+        }));
+        setFormData(form_value);
+    }, []);
+
+    React.useEffect(() => {
+        if (should_move_to_next) displayNextPage();
+    }, [should_move_to_next]);
+
+    const displayNextPage = () => {
+        if (form_data.risk_tolerance === 'No') {
+            // onSubmit hold reference to a function that takes 3 params - values, action and should_override
+            onSubmit(form_data, null, true);
+        } else {
+            const next_question = current_question_details.current_question_index + 1;
+            if (next_question < assessment_questions.length) {
+                setCurrentQuestionDetails({
+                    current_question_index: next_question,
+                    current_question: assessment_questions[next_question],
+                });
+            }
+        }
+    };
+
+    const displayPreviousPage = () => {
+        const prev_question = current_question_details.current_question_index - 1;
+        if (prev_question >= 0) {
+            setCurrentQuestionDetails({
+                current_question_index: prev_question,
+                current_question: assessment_questions[prev_question],
+            });
+        }
+    };
+
+    const handleOnClickNext = values => {
+        if (!isAssessmentCompleted(values)) {
+            displayNextPage();
+        }
+    };
+
+    const handleValueSelection = (e, form_control, callBackFn) => {
+        if (typeof e.persist === 'function') e.persist();
+        callBackFn(form_control, e.target.value);
+        setFormData(prev_form => ({ ...prev_form, [form_control]: e.target.value }));
+    };
+
+    const hideElement = condition => {
+        return condition ? { visibility: 'hidden' } : {};
+    };
+
+    const isAssessmentCompleted = answers => Object.values(answers).every(answer => Boolean(answer));
+
+    return (
+        <div className={classNames('trading-assessment', class_name)}>
+            <Text as='p' color='prominent' size='xxs' className='trading-assessment__side-note'>
+                <Localize i18n_default_text='In providing our services to you, we are required to obtain information from you in order to assess whether a given product or service is appropriate for you.' />
+            </Text>
+            <section className='trading-assessment__header'>
+                <div className='trading-assessment__header--background'>
+                    <Button
+                        onClick={displayPreviousPage}
+                        transparent
+                        style={hideElement(
```

Comment:
are you sure it works here? as I see <Button/> has no props style

Reply:
Button component takes `...prop`. That is how additional HTML attributes gets applied to html component

Reply:
yes. ok. but why not just use the props which are defined for our custom component, isn’t it more explicit and understandable ?